File: python/1328_Break_a_Palindrome.py
class Solution:
    def breakPalindrome(self, palindrome: str) -> str:
        n = len(palindrome)
        if n == 1:
            return ""

        def isPalindrom(palindrome:str) -> bool:
            i,j = 0,len(palindrome)-1
            while i <= j:
                if palindrome[i] == palindrome[j]:
                    i += 1
                    j -= 1
                else:
                    return False
            return True
        
        res = ""
        for i in range(n//2 + 1):
            if palindrome[i] != 'a':
                res = palindrome[:i] + 'a' + palindrome[i+1:]
                if not isPalindrom(res):
                    break
            if i == n//2:
                res = palindrome[:-1] + 'b'
                break
        if isPalindrom(res):
            return ""
        return res

# method 2
    def breakPalindrome1(self, palindrome: str) -> str:
        n = len(palindrome)
        if n <= 1:
            return ""
        palindrome = list(palindrome)

        for i in range(n//2):
            if palindrome[i] != 'a':
                palindrome[i] = 'a'
                return ''.join(palindrome)
        palindrome[-1] = 'b'
        return ''.join(palindrome)

File: java/1328_Break_a_Palindrome.java
class Solution {
    public String breakPalindrome(String palindrome) {
        int n = palindrome.length();
        String ans = "";
        for (int i = 0 ; i < n/2 ; i++){
            if(palindrome.charAt(i) != 'a'){
                ans = palindrome.substring(0,i) + 'a' + palindrome.substring(i+1,n);
                break;
            }
            if(i == n/2 -1){
                ans = palindrome.substring(0,n-1) + 'b';
                break;
            }
        }
        if(isPalindrom(ans)){
            return "";
        }
        return ans;
    }

    public static boolean isPalindrom(String s){
        return new StringBuilder(s).reverse().toString().equals(s);
    }
}